src/diffusers/training_utils.py

```diff
@@ -194,6 +195,7 @@ def __init__(
             inv_gamma (float):
                 Inverse multiplicative factor of EMA warmup. Default: 1. Only used if `use_ema_warmup` is True.
             power (float): Exponential factor of EMA warmup. Default: 2/3. Only used if `use_ema_warmup` is True.
+            foreach (bool): Use torch._foreach functions for updating shadow parameters. Should be faster.
```

Comment:
"Should be faster" reads a little brute-forced. Maybe provide documentation or resource links here?

Reply:
I'm mimicking the language used for this in torch optimizers -- "[usually significantly more performant](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html)".  It's on the order of milliseconds either way and is usually not going to be a huge bottleneck unless doing very quick training steps.  The operators themselves also do not have their own documentation, but have been used in all built-in torch optimizers as default behavior for a while now.  It is faster in my experiences testing it, though I do need to test this exact implementation since I have been using a slightly different EMA model for my own testing.